RecoLocalTracker/Phase2TrackerRecHits/test/RecHitsValidation.cc

```diff
@@ -0,0 +1,276 @@
+#include <iostream>
+#include <map>
+#include <vector>
+#include <algorithm>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/CommonDetUnit/interface/GeomDetUnit.h"
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/SiPixelDetId/interface/PixelSubdetector.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Phase2TrackerCluster/interface/Phase2TrackerCluster1D.h"
+#include "DataFormats/Phase2TrackerRecHit/interface/Phase2TrackerRecHit1D.h"
+
+#include "SimDataFormats/TrackerDigiSimLink/interface/PixelDigiSimLink.h"
+#include "SimDataFormats/Track/interface/SimTrackContainer.h"
+#include "SimDataFormats/Vertex/interface/SimVertexContainer.h"
+#include "SimDataFormats/TrackingHit/interface/PSimHitContainer.h"
+
+#include "DataFormats/GeometrySurface/interface/LocalError.h"
+#include "DataFormats/GeometryVector/interface/LocalPoint.h"
+
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "CommonTools/Utils/interface/TFileDirectory.h"
+
+#include <TH2D.h>
+#include <TH1D.h>
+#include <THStack.h>
+
+
+struct RecHitHistos {
+    THStack* numberRecHitsMixed;
+    TH1D* numberRecHitsPixel;
+    TH1D* numberRecHitsStrip;
+
+    TH2D* globalPosXY[3];
+    TH2D* localPosXY[3];
+};
+
+class Phase2TrackerRecHitsValidation : public edm::EDAnalyzer {
+
+    public:
+
+        explicit Phase2TrackerRecHitsValidation(const edm::ParameterSet&);
+        ~Phase2TrackerRecHitsValidation();
+        void beginJob();
```

Comment:
Methods like beginJob is something you'd expect to be called by the framework when needed. However, if the base class method changes name or signature, your function defined here may never be called.
Add  ``` override ``` to tell the compiler this has to be also in the base class.

the ED\* workers interface changed a few times in the past years. So, better learn early that the method you implemented actually does something in run time.